CD v1.39.x RN (#6843)
harness/developer-hub: release-notes/continuous-delivery.md

## May 2024

### Version 1.39.4

#### Fixed issues

> - Fixed an issue where the MS Teams notifications didn't show the event names. Now, the pipeline end event appears as "ended" and pipeline success event appears as "succeeded" in notifications. (PIPE-18855, ZD-62684)

Reviewer: - Fixed an issue where the MS Teams notifications didn't show the event names. Now, the pipeline end event appears as "ended" and the pipeline success event appears as "succeeded" in notifications. (PIPE-18855, ZD-62684)

> - Fixed an issue where the MS Teams notifications didn't show the event names. Now, the pipeline end event appears as "ended" and pipeline success event appears as "succeeded" in notifications. (PIPE-18855, ZD-62684)

Author: ```suggestion
- Fixed an issue where the MS Teams notifications didn't show the event names. Now, the pipeline end event appears as "ended" and the pipeline success event appears as "succeeded" in notifications. (PIPE-18855, ZD-62684)
```

> - For a pipeline with bidirectional sync enabled, the file name change in GitHub was not properly reflected in the Harness UI. This issue is fixed by enhancing the cache handing for files. (PIPE-18828, ZD62791)

Reviewer: - For a pipeline with bidirectional sync enabled, the file name change in GitHub was not properly reflected in the Harness UI. This issue is fixed by enhancing the cache handling for files. (PIPE-18828, ZD62791)

> - For a pipeline with bidirectional sync enabled, the file name change in GitHub was not properly reflected in the Harness UI. This issue is fixed by enhancing the cache handing for files. (PIPE-18828, ZD62791)

Author: ```suggestion
- For a pipeline with bidirectional sync enabled, the file name change in GitHub was not properly reflected in the Harness UI. This issue is fixed by enhancing the cache handling for files. (PIPE-18828, ZD62791)
```

> - Step group templates did not appear when selecting Use Template on an Approval stage. This issue is fixed. You can now create a step group template with Approval stage, and use them as step entities in the Approval stage. (CDS-96930, ZD-63556)

Reviewer: - Step group templates did not appear when selecting Use Template on an Approval stage. This issue is fixed. You can now create a step group template with the Approval stage, and use them as step entities in the Approval stage. (CDS-96930, ZD-63556)

> - Step group templates did not appear when selecting Use Template on an Approval stage. This issue is fixed. You can now create a step group template with Approval stage, and use them as step entities in the Approval stage. (CDS-96930, ZD-63556)

Author: ```suggestion
- Step group templates did not appear when selecting Use Template on an Approval stage. This issue is fixed. You can now create a step group template with the Approval stage, and use them as step entities in the Approval stage. (CDS-96930, ZD-63556)
```

> - Step group templates did not appear when selecting USe Template on an Approval stage. This issue is fixed. You can now create a step group template with Approval stage type, and can use them in the Approval stage as step entities. (CDS-96930, ZD-63556)

Author: ```suggestion
- Step group templates did not appear when selecting Use Template on an Approval stage. This issue is fixed. You can now create a step group template with Approval stage type, and can use them in the Approval stage as step entities. (CDS-96930, ZD-63556)
```

- The HTTP step's Request Body field stretched horizontally to fit lengthy expressions. This issue is fixed. (CDS-96839, ZD-63344)
- Fixed an issue where users were unable to save YAML changes when creating a new infrastructure definition. (CDS-96396, ZD-62801, ZD-63997)

> - Terraform configuration for a monitored service returned the `500` response code. This issue occurred due to incorrect un-marshalling of Terraform configuration to the JSON object for the API request. This led to backend validations failing and causing the `500` response code. This issue is fixed by adding more validations/null-checks in the backend. (CDS-96374, ZD-62737)

Reviewer: null checks

> - Terraform configuration for a monitored service returned the `500` response code. This issue occurred due to incorrect un-marshalling of Terraform configuration to the JSON object for the API request. This led to backend validations failing and causing the `500` response code. This issue is fixed by adding more validations/null-checks in the backend. (CDS-96374, ZD-62737)

Author: ```suggestion
- Terraform configuration for a monitored service returned the `500` response code. This issue occurred due to incorrect un-marshalling of Terraform configuration to the JSON object for the API request. This led to backend validations failing and causing the `500` response code. This issue is fixed by adding more validations/null checks in the backend. (CDS-96374, ZD-62737)
```

> - Terraform configuration for a monitored service returned the `500` response code. This issue occurred due to incorrect un-marshalling of Terraform configuration to the JSON object for the API request. This led to failing of backend validations causing the `500` response code. This issue is fixed by adding more validations/null-checks in the backend. (CDS-96374, ZD-62737)

Author: ```suggestion
- Terraform configuration for a monitored service returned the `500` response code. This issue occurred due to incorrect un-marshalling of Terraform configuration to the JSON object for the API request. This led to backend validations failing and causing the `500` response code. This issue is fixed by adding more validations/null-checks in the backend. (CDS-96374, ZD-62737)
```

- The TerraformCloud Run step did not fail if the Terraform Apply step failed. This issue occurred because we were not checking the status of Terraform Apply after streaming logs. If Apply started, the step passed even if the Apply step did not complete. The step failed only when the Apply step failed to start, but not if it failed midway. This issue is fixed. The TerraformCloud Run step now fails if the Apply step fails midway, allowing users to use appropriate failure strategies. (CDS-96089, ZD-62254, ZD-63371)
- Fixed an issue where runtime expressions and values were not supported for Helm Values files in the Update GitOps Apps step. (CDS-96005)

> - The Resource Constraint pop-up didn't show the on-going execution details in the pipeline execution console view. This issue is fixed. (CDS-18831, ZD-62825)

Author: ```suggestion
- The Resource Constraint pop-up didn't show the ongoing execution details in the pipeline execution console view. This issue is fixed. (CDS-18831, ZD-62825)
```

> 

Reviewer: ongoing
